### ISSUE TITLE
FR: Add touch scroll to Audio/Midi Edit/Overview

### ISSUE THREAD

**Issue by ColRad** [ThingstoneGmbH/Track8#94]
It would be an intuitive way to quickly move around the timeline. 
It could allow us to remove the Audio/Midi Overview Encoder Scroll behavior that doesn't quite fit with the rest of the Encoder functions. 

Lets discuss.

**Comment by ColRad** [comment]
@MoebiusSt what is your view on that?

**Comment by MoebiusSt** [comment]
My interpretation when you say "TOUCH SCROLL" is the following.

- people can put their finger down somewhere and drag their finger a distance. This scrolls the timeline beneath the finger with some finetuned scalar so that it feels like the entire content moved exactly with the finger.

- you would NOT introduce a variable factor that would make the drag feel faster or slower, you'd aim to make the Touch-Drag always feel consistent with what lies beneath the finger. (While the speed of the Grid/Encoder SCROLL is dependent on the GRID-Setting (with Encoder-1 in Audio/Midi-Overview, or Encoder-2 on Audio/Midi Editors) the Finger TOUCH SCROLL would be dependant on the ZOOM-Factor.

- The user would zoom-out-and-drag to move faster or zoom-in and drag to move more precise.

- We only want to move on one axis (time) in most cases. Any vertical movement from the drag would be discarded. 

**First reaction:**

You would NOT replace the GRID-ENCODER-SCROLL. You've put a lot of effort and design in this from the start, you made it work. It is fast, easily learned and ingrained in muscle memory, it feels like fundamentally woven into the concept with how time and the BEAT/BAR grid works, how the selection processes work for entirely different functions. Even how the grid adapts to SLAVED TIME ... etc. It clicks in your head as ONE. So you would only ADD the new TOUCH SCROLL as a comfort function. (imho)

But as an addition in some views TOUCH SCROLL could be a wonderfully intuitive. Especially, since you can use it independant on the MODE you're in (VOL/PAN/SCROLL in Audio Overview) or (Channel/Offset/Scroll in Midi-Overview). Having to mode-change to SCROLL is indeed a pain point that Touch-Drag could alleviate.

**General Caveats:**

- for most people being right handed, you'd have the left hand on the ZOOM-Encoder and the right finger would want to drag. Meaning: hands would have to cross each other and cover most of the screen. So you would likely have to relocate the ZOOM-Encoder function (on all Views) and people would have to relearn.

- Wear and tear of the screen would increase. We would smear dust and grease over the glass a hundredfold.

- people would expect other gestures ingrained from using smartphones for decades in the EDIT-MODES: Pinch-Zoom, HOLD-AND-DRAG (on the left and right selection boundaries of START/END). Tap-to-select (a certain Midi Note), Drag-Select a rectangular selection around several MIDI Nodes ... etc. Limiting Touch to only the one-dimensional drag COULD cause frustration. Would you want to go all the way?


Particular inconsistencies to be addressed:

- In Audio/Midi-Overview there is currently no Zoom-Factor, since until now it wasn't necessary.  So Touch-Drag-Speed would be fixed here, probably to 1/1. Or you'd introduce ZOOM here, or you'd make Drag-Speed dependent on the Grid-Setting and have points 'whoosh' by beneath your finger at X-times the drag-speed (which, i think, would feel unintuitive). I don't think you would want to introduce ZOOM in Audio-Overview. It would complicate things. You could use "inertial scrolling" which takes a lot of fine-tuning if done well (not just simple trailing motion but also acceleration and a way to stop inertia (MUST!). Would Caching and Display keep up?

- Enter Audio-Edit Mode. The first Mode active is "VOLUME" (or PAN and TRACKFX). A finger drag is used to TOUCH-DRAW a curve. Drawing collides with the TOUCH SCROLL. So in order to keep TOUCH SCROLL consistent you'd have to make TOUCH-DRAW a SHIFT-Function (in all views with Drawing) ... but that's not good either because most people would like to draw with their right finger – which currently sits on the shift-Button forcing you to draw with your left finger. Also SHIFT-DRAW is already used as CLEAR-BRUSH (Great stuff that Clear Brush!!)

- In AUDIO-EDIT > EDIT- and NUDGE-mode it would work very well, though.

- Also in MIDI-EDIT > NOTE- and EDIT-Mode it would work very well. It would really alleviate pain points from the cumbersome CROSSHAIR selection that the X- Y- ENCODER SCROLL made necessary. You'd probably even make an exception and not discard the Y-Axis movement of the TOUCH SCROLL and use it to move the piano-role (NOTE) up and down too. But if you'd hang on to the central crosshair, then we'd have to Start a Touch-Drag somewhere on the screen and move the entire screen like a sheet of paper under an overhead projector, until a midi note sits under the crosshair. That's some relief, but not a revolutionary step up. Better would be to go all the way and also introduce Tap-to-select to MOVE the crosshair or selectable notes! But that will NOT be simple ... A moved crosshair will then break the GRID-ENCODER-SCROLL function, unless any ENCODER-SCROLL resets the crosshair back to center .

- In CC and PITCHBEND you'd have it collide with the DRAW-CURVE function again.

**Comment by ColRad** [comment]
I basically have the same view on that. 
I created this Feature Request because a user mentioned it. 
But if we think it through it becomes less appealing. 

The whole Track8 Encoder/Buttons first concept stems from my frustration of gear that heavily relies on a touchscreen. 
And most of these touchscreen functions just feel bad (midi editing for example). 

There are some places where the Touchscreen offers a big benefit over encoder/button operation, like the naming keyboard. 

And then there are the places where the Touch function seems like a no brainer (touch to scroll) but that opens pandoras box if we follow that through.

**Comment by MoebiusSt** [comment]
My inkling is that at least in MIDI EDITING you will sooner or later be forced to think through Touch. I know they feel bad, but they would not be worse than then the current system of "Magnadoodle" every single action with X-/Y- Encoder NAV... You#ll be forced either by repeated user requests or by feedback: "i never use the midi editor". "only use the midi editor if i must as a last resort"... "the device HAS a frickin touch screen, WHY is it not used in the MIDI Editor, ffs?!"

For scroll... yea, no easy solution presents itself. Shelve it.

**Comment by ColRad** [comment]
Yeah the detail edit functions on the Midi Editor are quite hard to use. 
I usually use the Editor to Step Sequence noted from a midi keyboard, and do maybe Velocity/Quantize adjustments via the edit region. And maybe delete a single note that I played wrong. 
For everything else I usually just cut the whole part and record it again.

**Comment by MoebiusSt** [comment]
(#101 "Snap to midi notes"-mode toggle (IMPLEMENTED IN 1.1.2) added some relief for the midi navigation.)

**Comment by ColRad** [comment]
Yeah lets put this aside for now.
